pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -111,6 +112,7 @@ func New(signer crypto.Signer, advertisableAddresser AdvertisableAddressResolver
 		receivedHandshakes:    make(map[libp2ppeer.ID]struct{}),
 		libp2pID:              ownPeerID,
 		logger:                logger,
+		metrics:               newMetrics(),
```

Comment:
exposing metrics from this package might be tricky... since the package is internal and metrics need to be registered inside the `node.go` file